matrix_sdk/src/http_client.rs

```diff
@@ -63,35 +64,14 @@ impl HttpClient {
             request.try_into_http_request(&self.homeserver.to_string(), access_token)?
         };
 
-        let url = &request.uri().to_string();
-
-        let request_builder = match method {
-            HttpMethod::GET => self.inner.get(url),
-            HttpMethod::POST => {
-                let body = request.body().clone();
-                self.inner
-                    .post(url)
-                    .body(body)
-                    .header(reqwest::header::CONTENT_TYPE, "application/json")
-            }
-            HttpMethod::PUT => {
-                let body = request.body().clone();
-                self.inner
-                    .put(url)
-                    .body(body)
-                    .header(reqwest::header::CONTENT_TYPE, "application/json")
-            }
-            HttpMethod::DELETE => {
-                let body = request.body().clone();
-                self.inner
-                    .delete(url)
-                    .body(body)
-                    .header(reqwest::header::CONTENT_TYPE, "application/json")
-            }
-            method => panic!("Unsupported method {}", method),
-        };
+        if let HttpMethod::POST | HttpMethod::PUT | HttpMethod::DELETE = *request.method() {
+            request.headers_mut().append(
+                http::header::CONTENT_TYPE,
+                HeaderValue::from_static("application/json"),
+            );
+        }
 
-        Ok(request_builder.send().await?)
+        Ok(self.inner.execute(request.try_into()?).await?)
```

Comment:
This method doesn't seem to exist for the WASM version of `reqwest` so the CI fails under WASM.

Reply:
Fixed two days ago in https://github.com/seanmonstar/reqwest/pull/989, seems to just require another `reqwest` release. I don't think there's a good workaround, should I maybe just update reqwest to a git dependency for now?

Reply:
I guess so, since we're already blocked by a bunch of git deps adding another one won't hurt that much.